azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -66,6 +66,22 @@ def create_from_shared_access_signature(cls, sas_token):
         pipeline = IoTHubPipeline(authentication_provider)
         return cls(pipeline)
 
+    @classmethod
+    def create_from_x509_certificate(cls, hostname, device_id, x509):
```

Comment:
Implement this on the AbstractIoTHubDeviceClient since this is only implemented for device right now.